app/livechat/client/views/app/tabbar/visitorInfo.js

```diff
@@ -84,7 +79,31 @@ Template.visitorInfo.helpers({
 					const customFields = Template.instance().customFields.get();
 					if (customFields) {
 						const field = _.findWhere(customFields, { _id });
-						if (field && field.visibility !== 'hidden') {
+						if (field && field.visibility !== 'hidden' && field.scope === 'room') {
+							fields.push({ label: field.label, value: livechatData[_id] });
+						}
+					}
+				}
+			}
+			return fields;
+		}
+	},
+
+	customVisitorFields() {
+		const fields = [];
+		let livechatData = {};
+		const user = Template.instance().user.get();
+		if (user) {
+			livechatData = _.extend(livechatData, user.livechatData);
+		}
+
+		if (!_.isEmpty(livechatData)) {
```

Comment:
Looking at the complexity of your algorithm, I'd say you can write it better.
For instance:

```
	...
	customRoomFields() {
		const customFields = Template.instance().customFields.get();
		if (!customFields || customFields.length === 0) {
			return [];              
		}

		const fields = [];
		const room = Template.instance().room.get();
		const { livechatData = {} }  = room || {};
		...
```

Reply:
of course! Thank you)